rsconnect/admin_main.py

```diff
@@ -0,0 +1,671 @@
+import logging
+import re
+import json
+from os.path import exists
+
+import click
+from click.types import StringParamType
+import semver
+
+from . import VERSION
+from . import api
+from .actions import (
+    cli_feedback,
+    set_verbosity,
+)
+from .metadata import ServerStore
+from .models import AppModes, RebuildStatus
+
+# todo: instead of checking these for every command, we could just do this to skip the connection checks:
+# real_server, api_key, insecure, ca_data, from_store = server_store.resolve(name, url, api_key, insecure, ca_data)
+from .main import _validate_deploy_to_args
+
+from .admin_actions import (
+  open_file_or_stdout,
+  download_bundle,
+  rebuild_add_content,
+  rebuild_remove_content,
+  rebuild_list_content,
+  rebuild_history,
+  rebuild_start,
+  search_content,
+  get_content,
+  emit_rebuild_log,
+)
+
+
+server_store = ServerStore()
+future_enabled = False
+logging.basicConfig()
+
+_version_search_pattern = r"(^[=><]{1,2})(.*)"
+
+def _verify_rebuild_rm_args(guid, all, purge):
+    if guid and all:
+        raise api.RSConnectException("You must specify only one of -g/--guid or --all, not both.")
+    if not guid and not all:
+        raise api.RSConnectException("You must specify one of -g/--guid or --all.")
+
+@click.group(no_args_is_help=True)
+@click.option("--future", "-u", is_flag=True, hidden=True, help="Enables future functionality.")
```

Comment:
I like the concept of the `--future` option - what functionality does it expose?

Reply:
None yet. I shamelessly stole this from our existing main.py. Not needed by admin_main.py though, so it can be removed